python/cuml/ts/arima.pyx

```diff
@@ -0,0 +1,814 @@
+        llb = ll_f(self.num_batches, self.num_samples, self.order[0], self.y, x)
+        return [-2 * lli + np.log(len(self.y)) * (_model_complexity(self.order[i]))
+                for (i, lli) in enumerate(llb)]
+
+    @property
+    def aic(self):
+        (p, d, q) = self.order[0]
+        x = pack(p, d, q, self.num_batches, self.mu, self.ar_params, self.ma_params)
+        llb = ll_f(self.num_batches, self.num_samples, self.order[0], self.y, x)
+        return [-2 * lli + 2 * (_model_complexity(self.order[i]))
+                for (i, lli) in enumerate(llb)]
+
+    def _assert_same_d(self, b_order):
+        """Checks that all values of d in batched order are same"""
+        b_d = [d for _, d, _ in b_order]
+        assert (np.array(b_d) == b_d[0]).all()
+
+    def predict_in_sample(self):
+        """Return in-sample prediction on batched series given batched model
+
+        Returns:
+        --------
+        y_p : array-like, shape = (n_samples, n_series)
+
+        Example:
+        --------
+        .. code-block:: python
+            from cuml.ts.arima import fit
+            ...
+            model = fit(ys, (1,1,1), mu0, ar0, ma0)
+            y_pred = model.predict_in_sample()
+        """
+
+        p, d, q = self.order[0]
+        
+        handle = cuml.common.handle.Handle()
+        cdef cumlHandle* handle_ = <cumlHandle*><size_t>handle.getHandle()
+
+        x = pack(p, d, q, self.num_batches, self.mu, self.ar_params, self.ma_params)
+        cdef uintptr_t d_params_ptr
+        d_params, d_params_ptr, _, _, _ = input_to_dev_array(x, check_dtype=np.float64)
+
+        cdef np.ndarray[double, ndim=2, mode="fortran"] y_p = np.zeros(((self.num_samples),
+                                                                        self.num_batches), order="F")
+        cdef uintptr_t d_y_p_ptr
+        d_y_p, d_y_p_ptr, _, _, _ = input_to_dev_array(y_p, check_dtype=np.float64)
+
+        cdef np.ndarray[double, ndim=2, mode="fortran"] vs = np.zeros(((self.num_samples - d),
+                                                                       self.num_batches), order="F")
+
+        cdef uintptr_t d_vs_ptr
+        d_vs, d_vs_ptr, _, _, _ = input_to_dev_array(vs, check_dtype=np.float64)
+
+        cdef uintptr_t d_y_ptr    
+        if self.d_y is None:
+            d_y, d_y_ptr, _, _, _ = input_to_dev_array(self.y, check_dtype=np.float64)
+            self.d_y = (d_y, d_y_ptr)
+        
+        (_, d_y_ptr) = self.d_y
+
+        predict_in_sample(handle_[0], <double*>d_y_ptr,
+                          self.num_batches, self.num_samples, p, d, q,
+                          <double*>d_params_ptr, <double*>d_vs_ptr, <double*>d_y_p_ptr)
+
+        
+        
+        updateHost(&vs[0,0], <double*>d_vs_ptr, (self.num_samples-1) * self.num_batches, 0)
+        updateHost(&y_p[0,0], <double*>d_y_p_ptr, (self.num_samples) * self.num_batches, 0)
+
+        self.yp = y_p
+        return y_p
```

Comment:
Return objects from cuML are typically expected to be device data, so if `y` or `y_p` are rmm device arrays to start with, this would return device data as expected.

Reply:
I've updated `predict_in_sample` to return a device array.